downloads/swan-lake-release-notes/swan-lake-beta4/RELEASE_NOTE.md

```diff
@@ -0,0 +1,827 @@
+	
+    Prior to Swan Lake Beta4, the `error:stackTrace()` function returned an `error:CallStack` object which had the structure below which was a deviation from the specification.
+
+    ```ballerina
+    public class CallStack {
+    public CallStackElement[] callStack = [];
+    }
+    ```
+
+    This has now been fixed, and it is no longer possible to retrieve an `error:CallStack` object or directly access the `callStack` array as shown below.
+
+    ```ballerina
+    error:CallStack callStack = err.stackTrace(); // `CallStack` is undefined
+    error:CallStackElement[] elements = err.stackTrace().callStack; // Not allowed, `CallStackElement` is undefined, no `callStack` field.
+    ```
+
+    The `error:stackTrace` function now returns an array of `StackFrame` objects.
+
+    ```ballerina
+    public type StackFrame readonly & object {
+    public function toString() returns string;
+    };
+    ```
+
+    ```ballerina
+    error:StackFrame[] stackTrace = e.stackTrace(); // Now returns `error:StackFrame[]`.
+    ```
+
+- Made the return type of the `error:detail` function in the `lang.error` module a subtype of `readonly`. It is the intersection of `readonly` and the detail type of the error.
+
+    ```ballerina
+    type Detail record {|
+    int code;
+    |};
+    
+    type Error error<Detail>;
+    
+    function fn(Error e) {
+    Detail & readonly detail = e.detail(); // Now allowed.
+    }
+    ```
+
+- Fixed a deviation in the `stream:next` function’s stream argument name. The name has been changed from `strm` to `stm`.
+
+- Fixed a bug in `array:sort`, which was sorting the original list. The function now returns a new sorted array. The original array remains unchanged.
+
+- Changed the name of the argument to `transaction:setData()` from `e` to `data`. Moreover, changed the static type of the argument to `transaction:setData()` and the return type of `lang.transaction:getData()` to `readonly`. They were previously of type `(any|error) & readonly` and even this change would accept/return the same set of values.
+
+- Updated the `float:min()` and `float:max()` functions to return `float:NaN` if an argument is `float:NaN`. 
+
+    ```ballerina
+    import ballerina/io;
+
+    public function main() {
+        float result = float:min(1, float:NaN);
+        io:println(result === float:NaN); // prints true
+
+        result = float:max(5, float:NaN);
+        io:println(result === float:NaN); // prints true
+    }
+    ```
+
+- Fixed a bug in the `decimal:fromString()` function, which allowed parsing a string that matched the `HexFloatingPointLiteral`. It now returns an error.
+
+    ```ballerina
+    import ballerina/io;
+
+    public function main() {
+    decimal|error result = decimal:fromString("0xab12"); // Now returns an error.
+    io:println(result is error); // prints true
+    }
```

Comment:
This code block is not properly formatted

Reply:
Fixed.